dialog-el.html

```diff
@@ -55,102 +50,57 @@
-    }
-
-    :host([arrow-direction=right]) {
-      transform: translate(-50px,0);
-    }
-
-    :host([arrow-direction=top]) {
-      transform: translate(0,25px);
-    }
-
-    :host([arrow-direction=bottom]) {
-      transform: translate(0,-50px);
+    :host(:not([arrow-direction])) .pointer, 
+    :host([arrow-direction][modal]) .pointer {
+      display: none !important;
     }
 
     :host([arrow-direction]) .pointer {
-      height: 0px;
-    }
-
-    :host([arrow-direction]) .pointer:after {
-      border: 16px solid transparent;
-      content: '';
-      height: 0;
-      overflow: visible;
+      height: 22px;
+      width: 22px;
+      display: block;
       position: absolute;
-      width: 0;
-      opacity: 0;
-    }
-
-    :host([arrow-direction][visible]) .pointer:after {
-      opacity: 1;
-    }
-
-    :host([arrow-direction=left]) .pointer:after {
-      border-left-width: 0;
-      border-right-color: #FFF;
-      top: -16px;
-      left: -15px;
-    }
-
-    :host([arrow-direction=right]) .pointer:after {
-      border-left-color: #FFF;
-      border-right-width: 0;
-      left: -17px;
-      top: -16px;
-    }
-
-    :host([arrow-direction=top]) .pointer:after {
-      border-bottom-color: #FFF;
-      border-top-width: 0;
-      left: -18px;
-      top: -16px;
-    }
-
-    :host([arrow-direction=bottom]) .pointer:after {
-      border-bottom-width: 0;
-      border-top-color: #FFF;
-      left: -18px;
-      top: 16px
+      top: 0;
+      left: 50%;
+      transform: rotate(45deg) translate3d(-50%,-5px,0);
+      z-index: -1;
+      background-color: var(--dialog-el-background, #FFFFFF);
+      box-shadow: 0px 0px 0px 0px rgba(0,0,0,0.35), -1px -1px 1px 0px rgba(0,0,0,0.18);
     }
 
-    :host([arrow-direction=left]) .content {
-      -webkit-transform: translate(0, -40%) !important;
-              transform: translate(0, -40%) !important;
+    :host([arrow-direction=left]) .pointer {
+      top: 50%;
+      left: 0;
+      transform: rotate(45deg) translate3d(-14px,2px,0);
+      box-shadow: 0px 0px 0px 0px rgba(0,0,0,0.35), -1px 1px 1px 0px rgba(0,0,0,0.18);
     }
 
-    :host([arrow-direction=right]) .content {
-      right: 16px;
-      -webkit-transform: translate(-100%, -40%) !important;
-              transform: translate(-100%, -40%) !important;
+    :host([arrow-direction=bottom]) .pointer {
+      top: 100%;
+      left: 50%;
+      transform: rotate(45deg) translate3d(-50%,-5px,0);
+      box-shadow: 0px 0px 0px 0px rgba(0,0,0,0.35), 2px 2px 1px 0px rgba(0,0,0,0.18);
     }
 
-    :host([arrow-direction=bottom]) .content {
-      top: 16px;
-      -webkit-transform: translate(-50%, -100%) !important;
-              transform: translate(-50%, -100%) !important;
+    :host([arrow-direction=right]) .pointer {
+      top: 50%;
+      left: 100%;
+      transform: rotate(45deg) translate3d(-14px,2px,0);
+      box-shadow: 0px 0px 0px 0px rgba(0,0,0,0.35), 2px -1px 1px 0px rgba(0,0,0,0.18);
```

Comment:
Although the arrow is getting put in the correct place in relation to the dialog, the code that was removed was meant to move the entire dialog so that the arrow lines up with the dialog's original top and left position. That's what makes it appear as though it's pointing to whatever opened it.